django/db/models/sql/query.py

```diff
@@ -2521,23 +2501,11 @@ def trim_start(self, names_with_path):
         self.set_select([f.get_col(select_alias) for f in select_fields])
         return trimmed_prefix, contains_louter
 
-    def is_nullable(self, field):
-        """
-        Check if the given field should be treated as nullable.
-
-        Some backends treat '' as null and Django treats such fields as
-        nullable for those backends. In such situations field.null can be
-        False even if we should treat the field as nullable.
-        """
-        # We need to use DEFAULT_DB_ALIAS here, as QuerySet does not have
-        # (nor should it have) knowledge of which connection is going to be
-        # used. The proper fix would be to defer all decisions where
-        # is_nullable() is needed to the compiler stage, but that is not easy
-        # to do currently.
-        return field.null or (
-            field.empty_strings_allowed
-            and connections[DEFAULT_DB_ALIAS].features.interprets_empty_strings_as_nulls
-        )
+    def is_nullable(self, nullable_aliases):
+        # A subquery can always return no rows and thus be nullable but the ORM
+        # currently has some expectations with regards to IN vs ANY lookups that
+        # must be revisited to allow this logic to work properly.
+        return False
```

Comment:
If we wanted to enable this we'd have to change our usage of `field IN (subquery)` to `field = ANY (subquery)`, see ticket-20024 which is related.